src/test/expected/float-nan/main.rs

```diff
@@ -6,15 +6,17 @@ fn main() {
         f -= 1.0;
     }
 
-    // at this point, f == 0.0
-    // should succeed
-    assert!(1.0 / f != 0.0 / f);
-    // should succeed
-    assert!(!(1.0 / f == 0.0 / f));
-    // should fail
-    assert!(1.0 / f == 0.0 / f);
-    // should fail
-    assert!(0.0 / f == 0.0 / f);
-    // should suceed
-    assert!(0.0 / f != 0.0 / f);
+    match rmc::nondet::<u8>() {
+        // at this point, f == 0.0
+        // should succeed
+        0 => assert!(1.0 / f != 0.0 / f),
+        // should succeed
+        1 => assert!(!(1.0 / f == 0.0 / f)),
+        // should fail
+        2 => assert!(1.0 / f == 0.0 / f),
+        // should fail
+        3 => assert!(0.0 / f == 0.0 / f),
+        // should suceed
+        _ => assert!(0.0 / f != 0.0 / f),
+    }
```

Comment:
would it make sense to have a macro for this pattern

Reply:
I figured it wasn't worth the effort. There are a few differences in them and a macro would probably make it harder to read.

Reply:
I wonder if we should have the original, primitive cbmc assert available still.

Then our tests could still be straight-line code, just using `rmc::assert()` instead of `assert!`

Reply:
I was thinking about adding a function rmc::check() that had the behaviour you mentioned. What do you think? I would prefer not having two types of assert.

That said, I would prefer adding it in a different PR.

Reply:
Also, once we have the `#[proof]` attribute integrated, we should be able to easily split tests like this into multiple harnesses in the same file.

Reply:
I'm cool with a different name (brainstorming: `rmc::probe`? `check` also seems good) and in a follow up PR.

Reply:
I created https://github.com/model-checking/rmc/issues/695 to capture the work.